### PR TITLE
Add openjph

### PR DESCRIPTION
**Pull request by hmaarrfk**
Checklist
- [ ] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [ ] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [ ] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [ ] Build number is 0.
- [ ] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hmaarrfk** [comment]
@wolfv can you have a quick look here to ensure I got the basics down.

Are these recipes being accepted yet at conda-forge?

File: recipes/openjph/build.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+set -ex
+
+mkdir -p build
+pushd build
+
+# OJPH_BUILD_EXECUTABLES would require libtiff which creates a circular
```
Comment (wolfv) [inline recipes/openjph/build.sh:6]: This comment is outdated? :)

File: recipes/openjph/build.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+# dependency
+cmake ${CMAKE_ARGS}                          \
+    -DOJPH_BUILD_EXECUTABLES=ON              \
+    ..
+
+make -j ${CPU_COUNT}
```
Comment (wolfv) [inline recipes/openjph/build.sh:12]: I recommend ninja everywhere, but up to you!

File: recipes/openjph/recipe.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+tests:
+  - package_contents:
+      include:
+        - openjph/ojph_version.h
+  - script:
+    - if: unix
+      then:
+        - test -f ${PREFIX}/include/openjph/ojph_version.h
+        - test -f ${PREFIX}/lib/libopenjph${SHLIB_EXT}
```
Comment (wolfv) [inline recipes/openjph/recipe.yaml:39]: You could add these tests to package contents as well (although no harm to test them twice for the time being):

```
bin:
  - ojph_compress
  - ojph_expand
lib:
  - openjph
```

**Comment by wolfv** [comment]
@hmaarrfk I left a couple suggestions, looks good! The recipes are accepted. The main caveat is that the auto-tick bot doesn't work yet, but we'll try to fix that soon!

File: recipes/openjph/build.sh (hunk 1 above)
Comment (hmaarrfk) [reply recipes/openjph/build.sh]: yes, though i'm still curious about how to break that circular dependency in the future. these binaries leave alot to be desired.

File: recipes/openjph/build.sh (hunk 2 above)
Comment (hmaarrfk) [reply recipes/openjph/build.sh:10]: so much muscle memory.

File: recipes/openjph/recipe.yaml (hunk 3 above)
Comment (hmaarrfk) [reply recipes/openjph/recipe.yaml:39]: thanks.

File: recipes/openjph/recipe.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+        - which ojph_expand
+      else:
+        - where ojph_compress
+        - where ojph_expand
+
+about:
+  homepage: https://github.com/aous72/OpenJPH
+  license: BSD-2-Clause
+  license_file: LICENSE
+  summary: Open source implementation of High-throughput JPEG2000 (HTJ2K).
```
[the remark this answers is not in the document]
Comment (hmaarrfk) [reply recipes/openjph/recipe.yaml:50]: thanks.

**Comment by hmaarrfk** [comment]
> but we'll try to fix that soon!

Great ok. That will help me move things forward.

**Comment by wolfv** [comment]
OK, looks like the package content tests fail on Windows because the installed files are:


```
 │ │   - Library\bin\openjph.0.17.dll
...
 │ │   - Library\lib\openjph.0.17.lib

```

Which are in the correct locations but contain a "abi" specifier. For the moment you could skip this one on Windows by using:

```yaml
lib:
  - ${{ "openjph" if unix }}
```

And we'll figure out how to fix this. I am actually not sure if this is a valid pattern on Windows.

**Comment by hmaarrfk** [comment]
ok thanks! will start to merge other recipes that use v1 with the same caveat!

**Pull request merged**